offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/offergeneration/OfferTemplate.groovy

```diff
@@ -0,0 +1,370 @@
+            it.text(overheadPercentage)
+        }
+    }
+
+    private static void fillCustomerInformation(Document document, OfferContent offer) {
+        String customerName = String.format(
+                "%s %s %s",
+                offer.getCustomerTitle(),
+                offer.getCustomerFirstName(),
+                offer.getCustomerLastName())
+        document.getElementById("customer-name").text(customerName)
+        document.getElementById("customer-organisation").text(offer.getCustomerOrganisation())
+        document.getElementById("customer-street").text(offer.getCustomerStreet())
+        document.getElementById("customer-postal-code").text(offer.getCustomerPostalCode())
+        document.getElementById("customer-city").text(offer.getCustomerCity())
+        document.getElementById("customer-country").text(offer.getCustomerCountry())
+    }
+
+    private static void fillProjectManagerInformation(Document document, OfferContent offer) {
+        String projectManagerName = String.format(
+                "%s %s %s",
+                offer.getProjectManagerTitle(),
+                offer.getProjectManagerFirstName(),
+                offer.getProjectManagerLastName())
+        document.getElementById("project-manager-name").text(projectManagerName)
+        document.getElementById("project-manager-street").text(offer.getProjectManagerStreet())
+        document.getElementById("project-manager-postal-code").text(offer.getProjectManagerPostalCode())
+        document.getElementById("project-manager-city").text(offer.getProjectManagerCity())
+        document.getElementById("project-manager-email").text(offer.getProjectManagerEmail())
+    }
+
+    private static void fillOfferInformation(Document document, OfferContent offer) {
+        DateFormat dateFormat = DateFormat.getDateInstance(DateFormat.LONG, Locale.US)
+        document.getElementById("doctype-desc").text("Quotation")
+        document.getElementById("offer-date").text(dateFormat.format(offer.getCreationDate()))
+        document.getElementById("offer-identifier").text(offer.getOfferIdentifier())
+        document.getElementById("offer-expiry-date").text(dateFormat.format(offer.getExpirationDate()))
+        document.getElementById("project-title").text(offer.getProjectTitle())
+        document.getElementById("project-description").text(offer.getProjectObjective())
+        if (offer.getExperimentalDesign()) {
+            document.getElementById("experimental-design").text(offer.getExperimentalDesign())
+        }
+    }
+
+    private static void fillCostSummary(Document document, OfferContent offer) {
+        // the overhead costs
+        document.select(".overheads  .data-generation  > .costs").each {element ->
+            element.text(Currency.format(offer.getOverheadsDataGeneration()))
+        }
+        document.select(".overheads  .data-analysis  > .costs").each {element ->
+            element.text(Currency.format(offer.getOverheadsDataAnalysis()))
```

Comment:
Its misleading that we have a currency class that does formatting of the number but does not provide a currency per soe

Reply:
It does provide the currency with its constant field `SYMBOL`. The currency value formatting is done by the currency. The same could have been said about the previous solution with added complexity.